File: RedSheet.DbEntities/Models/LiteRelationshipRequire.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("LiteRelationshipRequires", Schema = "dbo")]
	[Serializable]
	public partial class LiteRelationshipRequire  
	{
 
	[Required] 
	[MaxLength(200)]
	public string ClassName { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int LiteRelationshipRequireId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string LiteRelationshipRequireName { get; set; } 
	[InverseProperty("LiteRelationshipRequire")]
	public ICollection<LiteProjectBackground> LiteProjectBackgrounds { get; set; } 


	public LiteRelationshipRequire()
	{
	this.LiteProjectBackgrounds = new HashSet<LiteProjectBackground>();

    }

    }
}

File: redsheet-portal/src/app/database-models/lite-relationship-require.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { LiteProjectBackground,  } from './'
export class LiteRelationshipRequire {
    constructor(liteRelationshipRequire?: LiteRelationshipRequire )  {
        let properties = [ "className", "liteRelationshipRequireId", "liteRelationshipRequireName", "liteProjectBackgrounds",];
        for (let property of properties)
            if (liteRelationshipRequire && liteRelationshipRequire[property])
                this[property] = liteRelationshipRequire[property];
    }
 
    @required()
    @maxLength(200)
	className : string =   undefined;
 
	liteRelationshipRequireId : number =   0 ;
 
    @required()
    @maxLength(200)
	liteRelationshipRequireName : string =   undefined;
	@nested(LiteProjectBackground)
	liteProjectBackgrounds: LiteProjectBackground[];



}
